logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/DeleteClientUseCase.kt

```diff
@@ -0,0 +1,28 @@
+package com.wire.kalium.logic.feature.client
+
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.client.ClientRepository
+import com.wire.kalium.logic.data.client.DeleteClientParam
+import com.wire.kalium.logic.functional.suspending
+
+interface DeleteClientUseCase {
+    suspend operator fun invoke(param: DeleteClientParam): DeleteClientResult
+}
+
+class DeleteClientUseCaseImpl(private val clientRepository: ClientRepository) : DeleteClientUseCase {
+    override suspend operator fun invoke(param: DeleteClientParam): DeleteClientResult = (suspending {
+        clientRepository.deleteClient(param)
+    }.fold({ failure ->
+        DeleteClientResult.Failure.Generic(failure)
+    }, {
+        DeleteClientResult.Success
+    }))!!
```

Comment:
not needed after the changes in #211 
```suggestion
    }))
```